addons/l10n_it_edi/models/ir_mail_server.py

```diff
@@ -156,7 +156,7 @@ def _create_invoice_from_mail(self, att_content_data, att_name, from_address):
         # Decode the file.
         # Use etree.XML because etree.fromstring doesn't accept string data starting with an encoding declaration
         try:
-            tree = etree.XML(att_content_data)
+            tree = etree.fromstring(att_content_data)
```

Comment:
do not we miss an encode? (and adjust the comment then also) @lordkrandel

Reply:
Forgot the comment, I'm fixing it.

The encode is called before the _create_invoice_from_mail call...
https://github.com/odoo/odoo/blob/1a3a0f9c44f442b9c03463d36ebf8fd4810661a0/addons/l10n_it_edi/models/ir_mail_server.py#L130

... it's there just because there is a second call to the _create_invoice_from_mail method when we have the zipped xmls, and it's done directly with bytes and requires no encode call.
https://github.com/odoo/odoo/blob/1a3a0f9c44f442b9c03463d36ebf8fd4810661a0/addons/l10n_it_edi/models/ir_mail_server.py#L187

see https://docs.python.org/3/library/zipfile.html#zipfile.ZipFile.read